lib/neynar/hasUserLiked.ts

```diff
@@ -0,0 +1,42 @@
+'use server';
+
+import { ReactionsResponse } from "@neynar/nodejs-sdk/build/neynar-api/v2";
+
+async function hasUserLikedCast(hash: string, fid: number) {
+  const options = {
+    method: 'GET',
+    headers: {
+      accept: 'application/json',
+      api_key: process.env.NEYNAR_API_KEY!
+    },
+  } as any;
+
+  const queryParams = new URLSearchParams({
+    hash,
+    types: 'likes',
+    limit: '100',
+  });
+
+  let cursor: string | undefined;
+
+  do {
+    const response = await fetch(
+      `https://api.neynar.com/v2/farcaster/reactions/cast?${queryParams}`,
+      options,
+    );
```

Comment:
this query seems like it won't work for casts with more than 100 likes right?
- if my like is `101` but you only get the first `100` likes, this system fails